src/test/scala/chiselTests/TransitNameSpec.scala

```diff
@@ -0,0 +1,56 @@
+package chiselTests
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import chisel3._
+import chisel3.experimental.RawModule
+import chisel3.util.TransitName
+
+import firrtl.FirrtlExecutionSuccess
+
+/** A NOT gate that inverts its input. */
+class MyModule extends RawModule {
+  val io = IO(new Bundle{})
+}
+
+/** A top-level module that instantiates three copies of "MyModule" */
+class Top extends RawModule {
+
+  /* The instance name will be named "MyModule" */
+  val foo = Module(new MyModule).io
+
+  /* The instance will be named "bar" */
+  val bar = {
+    val x = Module(new MyModule)
+    TransitName(x.io, x)
```

Comment:
This testcase implicitly relies on the return value of TransitName? (I had to think twice for a bit before I realized that `bar` became `x.io`.

Reply:
Yeah, `TransitName` is confusing in that it returns the `from` argument. I've clarified this with your above suggestion about adding documentation of the return type. Also, there are now better comments in the testcase.